examples/pubsub-authenticated.toit

```diff
@@ -7,56 +7,94 @@ import net
 import encoding.json
 import tls
 import net.x509
-import certificate_roots
+
+/**
+An example demonstrating how to subscribe and publish messages on
+  a broker, connected through TLS.
+
+This examples also demonstrates how to use MQTT username/password.
+
+Be default uses test.mosquitto.org. This server is publicly accessible and
+  shared among many users.
+*/
+
 
 HOST ::= "test.mosquitto.org"
-PORT ::= 8886
-SERVER_CERTIFICATE ::= certificate_roots.ISRG_ROOT_X1
+PORT ::= 8883
+
+// Alternatively, you could install the certificate_roots package
+// and use 'ISRG_ROOT_X1':
+//   `SERVER_CERTIFICATE ::= certificate_roots.ISRG_ROOT_X1`
+SERVER_CERTIFICATE ::= MOSQUITTO_ORG_CERTIFICATE
 
+CLIENT_ID ::= "toit-pubsub-$(random)"
 USERNAME ::= "rw"
 PASSWORD ::= "readwrite"
 
-CLIENT_ID ::= "toit-pubsub"
+TOPIC_PREFIX ::= "toit/pubsub-$(random)"
 
 main:
-  socket := net.open.tcp_connect HOST PORT
-
-  tls := tls.Socket.client socket
+  transport := mqtt.TcpTransport.tls net.open --host=HOST
     --root_certificates=[SERVER_CERTIFICATE]
 
-  client := mqtt.Client
-      CLIENT_ID
-      mqtt.TcpTransport tls
-      --username=USERNAME
-      --password=PASSWORD
+  client := mqtt.Client --transport=transport
+
+  options := mqtt.SessionOptions
+      --clean_session
+      --client_id = CLIENT_ID
```

Comment:
Why did you start using whitespace around `=` here?

Reply:
done, but I started using this syntax in other places.
We should decide of whether it's a good idea or not.
The idea was to avoid the confusion that `--foo=bar --gee` is equivalent to two parameters instead of just one.